web/modules/weather_data/src/Service/AFDParser.php

```diff
@@ -0,0 +1,371 @@
+        }
+
+        // See if this paragraph contains the end of the body
+        // characters, '$$'
+        $endOfBodyRegex = "/^\s*(?<eob>[$]{2})\s*/m";
+        if (preg_match($endOfBodyRegex, $currentString, $match)) {
+            $symbol = $match["eob"];
+            $this->updateCurrentContentType($symbol);
+            $currentString = preg_replace($endOfBodyRegex, "", $currentString);
+        }
+
+        // Parse the rest of the paragraph content with
+        // rules based on the parser's current set
+        // contentType.
+        // For example, if the content type is 'wwa',
+        // then we treat newlines differently than if
+        // it is 'generic'
+        if ($this->currentContentType == "preamble") {
+            $this->parsePreambleContent($currentString, $result);
+        } elseif ($this->currentContentType == "wwa") {
+            $this->parseWWAContent($currentString, $result);
+        } elseif ($this->currentContentType == "epilogue") {
+            $this->parseEpilogueContent($currentString, $result);
+        } else {
+            $this->parseGenericContent($currentString, $result);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Parse the "preamble" (before the first header)
+     * content into the appropriate node types.
+     * Typically, the first paragraph will contain
+     * document codes, and the second will contain
+     * WFO info text
+     */
+    public function parsePreambleContent(string $str, array &$result)
+    {
+        $nodeType = $this->preambleCount > 0 ? "preambleText" : "preambleCode";
+        $currentString = trim($str);
+        if ($currentString != "") {
+            array_push($result, [
+                "type" => $nodeType,
+                "content" => $currentString,
+            ]);
+        }
+        $this->preambleCount += 1;
+    }
+
+    /**
+     * Parse non-header paragraph text for the
+     * 'generic' content type.
+     */
+    public function parseGenericContent(string $str, array &$result)
+    {
+        $str = preg_replace("/\n/", " ", $str);
+        $str = ParsingUtility::normalizeSpaces($str);
+        $str = trim($str);
+        if ($str != "") {
+            array_push($result, [
+                "type" => "text",
+                "content" => $str,
+            ]);
+        }
+    }
+
+    /**
+     * Parse non-header paragraph text for
+     * the Watches/Warnings/Advisories
+     * ('wwa') contentType
+     */
+    public function parseWWAContent(string $str, array &$result)
+    {
+        // A newline followed by 4 spaces indentation
+        // indicates line continuation. Replace with the empty
+        // string, followed by a normal newline.
+        $currentString = trim($str);
+        if (preg_match("/SYNOPSIS/", $currentString)) {
+            $test = true;
+        }
+        $indentRegex = "/\n    +/";
+        $currentString = preg_replace($indentRegex, "", $currentString);
```

Comment:
I _think_ there is a test that covers it [here](https://github.com/weather-gov/weather.gov/pull/1563/files#diff-745f597f94c25e093b0ea5cb5201675774c9375738160c249246406bf27feb30R187-R193) but it's possible I've overlooked how this gets used and/or over-fit the case to one specific example

Reply:
Still haven't checked it locally (I'm working on it, I promise), but I think that test covers the epilogue text, which preserves the newlines rather than concatenates them into a single string? Also 100% open to the probability that I'm misreading this. 😅

Reply:
So I jammed in this text, to see how it would handle it:

![Screenshot 2024-08-21 at 9 47 46 AM](https://github.com/user-attachments/assets/98849935-bea9-40e5-a368-2f92848f4e43)

And this was the output:
![Screenshot 2024-08-21 at 10 26 29 AM](https://github.com/user-attachments/assets/56a8e1fe-2e3b-46c9-a5ed-541aa13e16e1)

So it looks like it is eating a space character. However, what makes it tricky is that the preceding line break ends with a `-` and so leaving out the space works well for that (`GMZ730-755-765-775` as an unbroken string). I dunno which is the better outcome or how likely either one is, though.